src/org/jetbrains/plugins/scala/codeInsight/intention/CreateCompanionObjectIntention.scala

```diff
@@ -0,0 +1,49 @@
+package org.jetbrains.plugins.scala.codeInsight.intention
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.jetbrains.plugins.scala.extensions.Parent
+import org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef.{ScClass, ScObject}
+import org.jetbrains.plugins.scala.lang.psi.impl.ScalaPsiElementFactory
+
+/**
+  * mattfowler
+  * 5/21/2016
+  */
+class CreateCompanionObjectIntention extends PsiElementBaseIntentionAction {
+  override def getText = "Create companion object for class"
+
+  override def invoke(project: Project, editor: Editor, psiElement: PsiElement): Unit = {
+    getClassIfAvailable(psiElement).foreach { clazz =>
+      val companion = ScalaPsiElementFactory.createObjectWithContext(
```

Comment:
Multiline strings have different line separators on different platforms. IDEA expects "\n", but on Windows it gets "\r\n".